### ISSUE TITLE
MariaDB is not supported with ANSI_QUOTES

### ISSUE THREAD

**Issue by vfouqueron** [petoju/terraform-provider-mysql#137]
Hi there,

please follow these steps to ensure quick resolution of your issue.

The maintainer (petoju) manages this project in a free time and thus doesn't have plenty of time to resolve all the issues. With that mentioned, PRs are always welcome.

### Provider version
3.0.55

### Terraform Configuration Files
In the ideal case, provide narrowed-down reproducer of your case. This should be a complete module with config connecting to localhost demonstrating the issue.

If you can, provide also docker command-line or docker-compose file starting mysql, which lead to the issue. That will remove any ambiguity regarding versions of tools and so on.

```hcl
resource "mysql_database" "app" {
  default_character_set = "utf8mb4"
  default_collation     = "utf8mb4_unicode_ci"
  name                  = "grafana-on-call"
}

resource "mysql_user" "grafana_on_call" {
  user               = "grafana-on-call"
  host               = var.nodes_url
  plaintext_password = random_password.mysql.result
  tls_option         = "SSL"
}

resource "mysql_grant" "grafana_on_call" {
  user       = mysql_user.grafana_on_call.user
  host       = mysql_user.grafana_on_call.host
  database   = mysql_database.app.name
  privileges = [
    "DELETE",
    "INSERT",
    "SELECT",
    "UPDATE"
  ]
}
```

### Debug Output
https://gist.github.com/vfouqueron/b9ef6a1b6494d3d64da5b9eaf842e764

### Expected Behavior
The created user get his grant created with success

### Actual Behavior
Terraform output this error:
```
| Error: Provider produced inconsistent result after apply
│ 
│ When applying changes to
│ module.applications.module.grafana-on-call.mysql_grant.grafana_on_call,
│ provider "provider[\"registry.terraform.io/petoju/mysql\"]" produced an
│ unexpected new value: Root object was present, but now absent.
│ 
│ This is a bug in the provider, which should be reported in the provider's
│ own issue tracker.
```

### Steps to Reproduce
1. `terraform apply`

**Comment by vfouqueron** [comment]
Thank you for your quick feedback. I'm using a MySQL 8 server

It works with 3.0.43 :) I'll stick with this version until you're able to fix it

**Comment by dgonzalezp** [comment]
Happening to me as well, thanks for updating @petoju. I tried version 3.0.43 and version 3.0.40 and the issue is still there. 
<img width="944" alt="image" src="https://github.com/petoju/terraform-provider-mysql/assets/20174436/862c1171-0168-4e04-95ac-0157e1aeee50">

Configuration files are similar to the ones provided, apply to an Azure MySQL flexible server ver 8.0

**Comment by petoju** [comment]
@dgonzalezp could you please send at least part of the log? Our description on getting the logs is in our issue template: https://github.com/petoju/terraform-provider-mysql/blob/master/.github/ISSUE_TEMPLATE.md

I believe your issue is completely different from the originally reported one.

**Comment by petoju** [comment]
@vfouqueron this should be fixed in provider 3.0.56

**Comment by dgonzalezp** [comment]
Issue was fixed for me too, can confirm it works with version 3.0.57, thanks @petoju for your great work
